rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyConstantIfBranchExecution.java

```diff
@@ -55,67 +58,88 @@ public J visitWhileLoop(J.WhileLoop whileLoop, ExecutionContext executionContext
         @Override
         public J visitBlock(J.Block block, ExecutionContext executionContext) {
             J.Block bl = (J.Block) super.visitBlock(block, executionContext);
-            List<Statement> addStatements = getCursor().pollMessage("statements");
-            J.If removeIf = getCursor().pollMessage("remove-if");
-            if (removeIf != null) {
-                bl = maybeAutoFormat(bl, bl.withStatements(ListUtils.flatMap(bl.getStatements(), stmt -> {
-                    if (stmt == removeIf) {
-                        return addStatements;
-                    }
-                    return stmt;
-                })), executionContext, getCursor().getParent());
+            bl = (J.Block) new RemoveUnneededBlock.RemoveUnneededBlockStatementVisitor()
+                .visitNonNull(bl, executionContext, getCursor().getParentOrThrow());
+            EmptyBlockStyle style = ((SourceFile) getCursor().firstEnclosingOrThrow(JavaSourceFile.class))
+                .getStyle(EmptyBlockStyle.class);
+            if (style == null) {
+                style = Checkstyle.emptyBlock();
             }
+            bl = (J.Block) new EmptyBlockVisitor<>(style)
+                .visitNonNull(bl, executionContext, getCursor().getParentOrThrow());
```

Comment:
these operations should be guarded to prevent modifications that are unrelated to the `SimplifyConstantIfBranchExecution` concerns